File: LeetCode_POTD/June_2026/June_8.java
package Leetcode_POTD.June_2026;
/*
Normal Question: Take element in order in another memory space
*/

import java.util.ArrayList;

public class June_8 {
    public int[] pivotArray(int[] nums, int pivot) { //To keep less memory consumption
        ArrayList<Integer> a1=new ArrayList<>();
        ArrayList<Integer>a2=new ArrayList<>();
        for(int i:nums){
            if(i<pivot)a1.add(i);
            else if(i>pivot)a2.add(i);
        }
        int i=0;
        for(int j:a1)nums[i++]=j;
        for(;i<nums.length-a2.size();i++) nums[i]=pivot;
        for(int j:a2) nums[i++]=j;
        return nums;
    }

    public int[] pivotArray1(int[] nums, int pivot) { //T keep lesser runtime(Preferred)
        int[] result = new int[nums.length];
        int left = 0, right = nums.length - 1;
        for (int i = 0, j = nums.length - 1; i < nums.length; i++, j--) {
            if (nums[i] < pivot) {
                result[left] = nums[i];
                left++;
            }
            if (nums[j] > pivot) {
                result[right] = nums[j];
                right--;
            }
        }
        while (left <= right) {
            result[left] = pivot;
            left++;
        }
        return result;
    }
}

File: LeetCode_POTD/June_2026/June_8.py
"""
Normal Question: Take element in order in another memory space

New Learning: Range(n) is evaluated once only, now even if you change value of n in loop, no impact
"""
class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]: #For less memory
        l=len(nums)
        i=0; j=l-1
        res=[None]*l
        for k in range(j+1):
            if nums[k]<pivot: 
                res[i]=nums[k]
                i+=1
            if nums[l-k-1]>pivot:
                res[j]=nums[l-k-1]
                j-=1
        while True: 
            res[i]=pivot
            if i==j: break;
            i+=1
        return res
    
    def pivotArray1(self, nums: List[int], pivot: int) -> List[int]: #For less time (Preferred)
        L1 = []
        L2 = []
        L3 = []
        for i in nums:
            if i<pivot:
                L1.append(i)
            elif i>pivot:
                L3.append(i)
            else:
                L2.append(i)
        return L1 + L2 + L3